### DISCUSSION TITLE
Great project, but do we need to update HA for all 317 UK councils incrementally??

### DISCUSSION THREAD

**Discussion by bashers222** [robbrad/UKBinCollectionData#1085] [Ideas]
Do we really need to update all the councils added update by update?
Could they not be broken down into modules to be pulled by the app from github depending on your preset location?

I'm getting asked by HA to update this plugin regularly to add councils i'm never going to live in, otherwise put up with permaent notifications.
Is there no alternative?

**Comment by dp247** [comment]
> Do we really need to update all the councils added update by update?

Yes, unfortunately that's how the Github repo works. We also provide multiple new councils / updates to existing councils / new features in a single pull request, which means that multiple files often get updated with each pull request. Since the HACS addon is just a mirror of the repo, that's why there's so many updates.

> Could they not be broken down into modules to be pulled by the app from github depending on your preset location?

Not sure. At this point, I'd say no since it would require a core module and then a separate repo for each council. We also don't have a way of defaulting a user location to a council, since all of the scripts work with different info (paon, postcode, uprn etc).

**Comment by robbrad** [comment]
Thank you @bashers222 - There is a single Python PIP package (https://pypi.org/project/uk_bin_collection/) that is built and released every time a council is added or fixed.

I don't know how else we would manage this :( 

The purpose of the integration is to provide a single, easy-to-use component, and with that comes a certain level of singularity. 

How would your installed version know that your council was not changed, etc.? And if it did and ignored updates, what happens when we need to fix vulnerabilities and your version needs updating?

The only way I can imagine reducing this is to move to monthly releases, but that would mean people waiting for their council to be added. Maybe that's the right trade-off?

It looks like the idea to ignore was raised on HACS: https://github.com/hacs/integration/issues/1580

**Comment by bashers222** [comment]
OK, it was worth asking.
Thanks for the great work though folks, especially handy over christmas bank holidays!!

**Comment by s9550110** [comment]
Moving to a monthly (or even just weekly) updates would be greatly appreciated. I seem to be restarting HA every other day just for a UKBinCollection update. I'm loving the integration by the way :-)
